pkg/antctl/raw/test/client.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+type Client struct {
+	Clientset   kubernetes.Interface
+	Config      *rest.Config
+	RawConfig   clientcmdapi.Config
+	contextName string
+}
+
+func NewClient(contextName, kubeconfig string) (*Client, error) {
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+
+	if kubeconfig != "" {
+		rules.ExplicitPath = kubeconfig
+	}
+
+	nonInteractiveClient := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, &clientcmd.ConfigOverrides{CurrentContext: contextName})
+
+	config, err := nonInteractiveClient.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	rawConfig, err := nonInteractiveClient.RawConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	if contextName == "" {
+		contextName = rawConfig.CurrentContext
+	}
+
+	return &Client{
+		Clientset:   clientset,
+		Config:      config,
+		RawConfig:   rawConfig,
+		contextName: contextName,
+	}, nil
+}
+
+func (c *Client) ContextName() (name string) {
+	return c.contextName
+}
+
+func (c *Client) ClusterName() (name string) {
+	if context, ok := c.RawConfig.Contexts[c.ContextName()]; ok {
+		name = context.Cluster
+	}
+	return
+}
```

Comment:
why not have a `clusterName` field in the `Client` struct instead? This way we only need to get it once, and we probably don't need to keep `RawConfig` in the struct either.

Reply:
great insights. Thanks :)